src/identity/manager.rs

```diff
@@ -232,6 +234,25 @@ impl Worker {
         // refresh. In other words, at any point in time, there are no high-priority
         // (not Background) items scheduled to run in the future.
         let mut pending: PriorityQueue<Identity, PendingPriority> = PriorityQueue::new();
+        // The backoff strategy used for retrying operations. Sets the initial values for the backoff.
+        // The values are chosen to be reasonable for the CA client to be able to recover from transient
+        // errors.
+        let mut cert_backoff = ExponentialBackoff {
+            initial_interval: Duration::from_millis(500),
+            current_interval: Duration::from_secs(1),
+            max_interval: Duration::from_secs(60),
+            // cert_backoff.max_elapsed_time is the maximum elapsed time after instantiating
+            // [`ExponentialBackoff`](https://docs.rs/backoff/0.4.0/backoff/type.ExponentialBackoff.html)
+            // or calling [`reset`](https://docs.rs/backoff/0.4.0/backoff/backoff/trait.Backoff.html#method.reset)
+            // after which [`next_backoff`](https://docs.rs/backoff/0.4.0/backoff/backoff/trait.Backoff.html#method.next_backoff)
+            // returns `None`.
+            //
+            // We set it to 5 minutes (300 seconds) to ensure that we don't retry indefinitely.
```

Comment:
Do we want to stop? I worry if we have a 5 min outage of the CA (not unreasonable) we will never recover?

Reply:
+1 - retry indefinitely but not too frequently

Reply:
Should we not have a max_elapsed_time at all and just retry indefinitely w/ no panic **or** should we pick a higher time, like 30 minutes (not sure if this is too high or unreasonable) and maybe even a slightly higher max_interval and lower initial_interval?

Reply:
Indefinitely with a 1-5min timeout (not sure exactly what, but something in that range?) seems reasonable. Anytime we 'give up' we would never recover which seems pretty bad

Reply:
I can remove the max_elapsed_time and changed max_interval to 2.5 minutes. The exponential retry logic will calculate new backoff intervals on error until it hits either a success or the max_interval. Once the max_interval is hit, it will continue to do 2.5 minute waits until a cert is successful which will trigger the backoff reset and the exponential calculations start back at the initial_interval next time around for each cert.


The backoff crates reset function that will be called:
```
fn reset(&mut self) {
    self.current_interval = self.initial_interval;
    self.start_time = self.clock.now();
}
```

Since we're setting refresh_at to 2.5 minutes in the worst case, that could mean that we'd still be waiting on some certs to finish their wait in the pending priority queue.

Wdyt of this @howardjohn @keithmattix ?

Reply:
2.5 minutes worst case in the queue is acceptable IMO ; even the k8s crashloopbackoff is higher I think